ptf_tests/common/utils/test_utils.py

```diff
@@ -1021,3 +1041,160 @@ def host_check_netperf(remote=False, hostname="",username="",password=""):
+                  scapy.TCP()/scapy.Raw(\'0\'*50)' %(eth_dst,eth_src,ip_dst,ip_src))
+
+
+    print('Sending scapy traffic on iface=\"%s\",count=%s' % (iface,num))
+    status = conn.sendCmd('scapy.sendp(pkt,iface=\"%s\",count=%s)' % (iface,num))
+    # status = conn.sendCmd(cmd)
+    if status:
+        print(f"Command: scapy send executed from {vm_name} ")
+    else:
+        print("Failed to execute command scapy send")
+
+    result = conn.readResult()
+    if result:
+        match = re.search('Sent (\d*) packets', result)
+        if match:
+            pkt_sent = int(match.group(1))
+
+    ##Killing python3 terminal
+    status = conn.sendCmd('exit()')
+    if not status:
+        print("Failed to exit from python prompt")
+
+    if f"Sent {num} packets" in result:
+        print(f"PASS: Sent {num} to dst_ip: {ip_dst}")
+        print(f"result:{result}")
+        return True
+    else:
+        print(f"FAIL: Sent Failed--> {result} sent, expected {num} packets to be sent to dst_ip: {ip_dst}")
+        return False
+
+
+def verify_scapy_traffic_from_vm(vm_id,conn,config_data):
+    """
+    To verify packets captured on pcap file at receiver vm
+    STEPS:
+    1. Verify if pcap file exists on receiver_vm
+    2. use scapy.sniff to decode packets on the pcap file
+    3. verify the pkt_count on pcap file == expected_pkt_count (count)
+    """
+
+    result = False
+    packets = []
+    count = config_data['traffic']['number_pkts'][0]
+    pcap_file = config_data['traffic']['pcap_file_name'][0]
+    vm = config_data['vm'][vm_id]
+    vm_name = vm['name']
+
+    send_ctrl_c(conn)
+    print(f"#######Receiving Scapy Traffic from the receiver_vm {vm_name} ########")
+
+    #STEP1 : Looking for the PCAP file on the receiver VM
+    print(f"#######STEP1: Looking for the pcap file {pcap_file} on rcv_vm {vm_name} ########")
+  
+    cmd_list = ['python3', "from scapy.all import *", 
+                'ls %s' % pcap_file]
+
+    for cmd in cmd_list:
+        conn.sendCmd(cmd)
+    if conn.readResult():
+        print(f"Pcap file {pcap_file} exists on {vm_name}")
+    else:
+        print("Pcap file {pcap_file} does not exist on {vm_name}")
+
+    #STEP2: Use scapy.sniff to decode packets on the pcap file
+    print(f"STEP2: Use scapy.sniff to decode packets on the pcap file {pcap_file}")
+    cmd_list = ['python3', "from scapy.all import *",
+                'packets=[x for x in sniff(offline=\"%s\")]' %pcap_file]
+    for cmd in cmd_list:
+        conn.sendCmd(cmd)
+
+    conn.sendCmd('out_pkts= [1 for pkt in packets if pkt.version==4]')
+    conn.sendCmd('print out_pkts ')
+    conn.sendCmd('len(out_pkts)')
+    out1 = conn.readResult()
+    out= out1.split('\n')
+    if out:
+        match = re.search('.*(\d+).*', out[-2])
+        if match:
+            pkt_count = int(match.group(0)[:-1])
+
+    print(f"Packets capture on receiver_vm {vm_name} are pkt_count: {pkt_count}")
```

Comment:
Need to decode multicast and broadcast traffic, not just get the count.  You can use filter in tcpdump to capture only multicast or broadcast traffic.

Reply:
conn.sendCmd('out_pkts= [1 for pkt in packets if pkt.version==4]')